### PR TITLE
Refactor CollectionAssociation#reset

### PR DESCRIPTION
**Pull request by jcoyne**

### CODE REVIEW

**Comment by cam156** [comment]
@jcoyne where are the tests?  What does the @stale_state = nil fix?

**Comment by jcoyne** [comment]
@cam156 no tests added because there was no new functionality added. The changed code is covered (see coveralls)

**Comment by jcoyne** [comment]
https://coveralls.io/builds/2940422/source?filename=lib%2Factive_fedora%2Fassociations%2Fcollection_association.rb#L49

**Comment by jcoyne** [comment]
https://coveralls.io/builds/2940422/source?filename=lib%2Factive_fedora%2Fassociations%2Fcollection_association.rb#L49

**Comment by grosscol** [comment]
Looks good.  Functionally `Array.new` and `= []` are equivalent.  Uses a call to super to reset instance variables: loaded, target, stale_state, and inversed.

File: lib/active_fedora/associations/association.rb [hunk 1]
```diff
@@ -24,6 +24,7 @@ def initialize(owner, reflection)
       def reset
         @loaded = false
         @target = nil
+        @stale_state = nil
```
Comment (cam156) [inline lib/active_fedora/associations/association.rb]: @jcoyne isn't this new?

**Pull request merged**